### PR TITLE
Add RHEL8 Builder for Azure

### PR DESCRIPTION
**Pull request by ferricoxide**
Addresses need for RHEL8 – but not CentOS 8-stream or other relatives – builders per #394 

* Declare variable "azure_custom_managed_image_name_rhel8"
* Declare variable "azure_custom_managed_image_resource_group_name_rhel8"
* Declare source "azure-arm.base"
* Declare a shell-provisioner that tolerates SELinux not being available
* Exempts the Azure-RHEL8 build from a shell provisioner that does not tolerate SELinux not being available

### CODE REVIEW

File: spel/minimal-linux.pkr.hcl [hunk 1]
```diff
@@ -550,6 +550,21 @@ variable "amigen8_storage_layout" {
   ]
 }
 
+variable "azure_custom_managed_image_name_rhel8" {
+  description = "Name of a custom managed image to use as the base image for RHEL7 builds"
```
Comment (lorengordon) [inline spel/minimal-linux.pkr.hcl:554]:
```suggestion
  description = "Name of a custom managed image to use as the base image for RHEL8 builds"
```

File: spel/minimal-linux.pkr.hcl [hunk 2]
```diff
@@ -980,6 +1021,41 @@ build {
...
+      "SPEL_CLOUDPROVIDER=azure",
+      "SPEL_EXTRARPMS=${local.amigen8_extra_rpms}",
+      "SPEL_FIPSDISABLE=${var.amigen_fips_disable}",
+      "SPEL_GRUBTMOUT=${var.amigen_grub_timeout}",
+      "SPEL_HTTP_PROXY=${var.spel_http_proxy}",
+      "SPEL_USEDEFAULTREPOS=${var.amigen_use_default_repos}",
+    ]
+    execute_command = "{{ .Vars }} sudo -E /bin/bash '{{ .Path }}'"
+    only = [
+      "azure-arm.minimal-centos-8-image",
```
Comment (lorengordon) [inline spel/minimal-linux.pkr.hcl:1051]: There is no centos-8 source for azure yet. This is a bit confusing.
```suggestion
```

Comment (ferricoxide) [reply spel/minimal-linux.pkr.hcl:1051]: Yeah. Was copy-paystah'ing @nortiz34's provisional code to create a more cleaned-up configuration. I'd blipped the other `…centos-8…` reference, guess I missed this one.

**Comment by ferricoxide** [comment]
Note: the testing-environment sits behind a web proxy. Didn't (previously) _think_ that it was a _caching_-proxy, but, right now, even though Packer is saying it's pulling fresh AMIgen8 content, what's executing says that it's actually pulling _stale_ AMIgen8 content.  I did a `git show` of the `PostBuild.sh` script against the remote that Packer _says_ it's pulling fresh from: the `git show` shows the correct content but the manner in which `PostBuild.sh` is _executing_ says that whatever Packer is pulling is stale. I _think_ that the host from which Packer is executed transits the same proxy as the Packer build-VM, so, there _shouldn't_ be any such differences, but, the supposition for the witnessed-behavior is all I got.

Going to try a new Packer run, tomorrow, and hope that things are a bit better behaved. If not, will need to further investigate possible sources for the seeming disconnect.

**Comment by lorengordon** [comment]
go codebuild go

**Review by lorengordon — APPROVED** [review]

**Pull request merged**